### DISCUSSION TITLE
SHELL process cat file report error, please fix

### DISCUSSION THREAD

**Discussion by suncf1985** [apache/dolphinscheduler#7821] [General]
When you use the cat command to view the text content, the content includes the following content, then the node is an error node, and it has nothing to do with the process at this time. Please verify

文件内容如下，就是一个正常的文本文件，该 文本为hive 或MR 产生的日志文件通过shell方式调用：
```
Starting Job = job_1596159072514_0070, Tracking URL = http://cloudera02:8088/proxy/application_1596159072514_0070/
Kill Command = /opt/cloudera/parcels/CDH-5.12.1-1.cdh5.12.1.p0.3/lib/hadoop/bin/hadoop job  -kill job_1596159072514_0070
Hadoop job information for Stage-1: number of mappers: 1; number of reducers: 0
[root@cloudera05 demo1]# vi tt.txt 
S22121tarting Job = job_1596159072514_0070, Tracking URL = http://cloudera02:8088/proxy/application_1596159072514_0070/
Kill Command = /opt/cloudera/parcels/CDH-5.12.1-1.cdh5.12.1.p0.3/lib/hadoop/bin/hadoop job  -kill job_1596159072514_0070
Hadoop job information for Stage-1: number of mappers: 1; number of reducers: 0
```

shell流程中输入如下:
![image](https://user-images.githubusercontent.com/16234399/103206145-21db1680-4936-11eb-9556-0a145852a557.png)

此时,运行此shell节点,会以失败返回。
![image](https://user-images.githubusercontent.com/16234399/103206191-3fa87b80-4936-11eb-81f3-2d9ff7f76cb4.png)

![image](https://user-images.githubusercontent.com/16234399/103206201-46cf8980-4936-11eb-87c3-d0ecf273306b.png)

**Comment by CalvinKirs** [comment]
For better global communication, Please describe it in English. If you feel the description in English is not clear, then you can append description in Chinese(just for Mandarin(CN)), thx!

**Comment by suncf1985** [comment]
SHELL process cat file report error, please fix

When you use the cat command to view the text content, the content includes the following content, then the node is an error node, and it has nothing to do with the process at this time. Please verify

**Comment by suncf1985** [comment]
Starting Job = job_1596159072514_0070, Tracking URL = http://cloudera02:8088/proxy/application_1596159072514_0070/
Kill Command = /opt/cloudera/parcels/CDH-5.12.1-1.cdh5.12.1.p0.3/lib/hadoop/bin/hadoop job  -kill job_1596159072514_0070
Hadoop job information for Stage-1: number of mappers: 1; number of reducers: 0

**Comment by vimacro** [comment]
Because the text words with “application_*”，The worker scan the output of the task, if  the application id matched, then try to get yarn application state 
![image](https://user-images.githubusercontent.com/11975398/103266985-b94e7100-49eb-11eb-9b74-db5623ff2955.png)

**Comment by suncf1985** [comment]
> Because the text words with “application_*”，The worker scan the output of the task, if the application id matched, then try to get yarn application state
> ![image](https://user-images.githubusercontent.com/11975398/103266985-b94e7100-49eb-11eb-9b74-db5623ff2955.png)

这个需要修改源码解决吗?这应该是一个BUG吧.hive或MR的日志都带这样的日志

**Comment by vimacro** [comment]
> > Because the text words with “application_*”，The worker scan the output of the task, if the application id matched, then try to get yarn application state
> > ![image](https://user-images.githubusercontent.com/11975398/103266985-b94e7100-49eb-11eb-9b74-db5623ff2955.png)
> 
> 这个需要修改源码解决吗?这应该是一个BUG吧.hive或MR的日志都带这样的日志

It need to modify the source code.

**Comment by lenboo** [comment]
We need to discuss whether there are yarn tasks by detecting logs and whether this mechanism needs to be changed
